tests/php/validation/test-class-amp-validation-manager.php

```diff
@@ -1182,6 +1188,19 @@ function ( $sources ) {
 	 * @param callable $assert   Function to assert the expected sources.
 	 */
 	public function test_locate_sources_e2e( $callback, $xpath, $assert ) {
+		// @todo Remove once https://github.com/WordPress/gutenberg/pull/23104 is in a release.
+		// Temporary fixes an issue with PHP errors being thrown in Gutenberg v8.3.0 on PHP 7.4.
+		if ( ! (
+				current_theme_supports( 'editor-color-palette' ) &&
+				current_theme_supports( 'editor-gradient-presets' ) &&
+				current_theme_supports( 'editor-font-sizes' )
+		) ) {
+			$args = [];
+			add_theme_support( 'editor-color-palette', $args );
+			add_theme_support( 'editor-gradient-presets', $args );
+			add_theme_support( 'editor-font-sizes', $args );
+		}
```

Comment:
Unless I'm misunderstanding the purpose of this logic here, if one or two of them are already set, but with a bool for example, this will still fail.

The logic needs to work on them individually.
```suggestion
		$theme_features = [
			'editor-color-palette',
			'editor-gradient-presets',
			'editor-font-sizes',
		];

		foreach ( $theme_features as $theme_feature ) {
			if ( ! current_theme_supports( $theme_feature ) ) {
				add_theme_support( $theme_feature, [] );
			}
		}
```